### PR TITLE
Include abundances

### PR DESCRIPTION
**Pull request by skrakau**
Compute protein weight based on taxonomic abundances or contig depths.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/metapep branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/metapep)

### CODE REVIEW

**Comment by skrakau** [comment]
... still need to do some code cleaning.

File: bin/download_proteins_entrez.py [hunk 1]
```diff
@@ -74,33 +78,49 @@ def get_assembly_length(assembly_id):
     return root.find("./Stats/Stat[@category='total_length'][@sequence_tag='all']").text
 
 
+# get protein weight based on taxonomic abundances (if occuring in multiple taxa -> sum)
+# NOTE multiple occurences of a protein within one assembly are not counted!
+def get_protein_weight(proteinId, dict_proteinId_assemblyIds, dict_taxId_assemblyid, dic_taxId_abundance):
+    weight = 0.0
+    for assembly in dict_proteinId_assemblyIds[proteinId]:
```
Comment (lkuchenb) [inline bin/download_proteins_entrez.py:85]: What is the outer for loop required for?

File: bin/download_proteins_entrez.py [hunk 2]
```diff
@@ -74,33 +78,49 @@ def get_assembly_length(assembly_id):
...
-    prot_assembly_id_output = "proteinid_assemblyids.txt"
-
-    with open(args.taxid_input, 'rt') as in_file:
-        taxids = [ row[0] for row in csv.reader(in_file) ]
+    # read taxonomic ids for download (together with abundances)
+    reader = csv.reader(args.taxid_input)
+    header = next(reader)
+    if header[0] != "taxid":
+        print("ERROR: header is ", header)
+        sys.exit("Format of taxid_input wrong!")
```
Comment (lkuchenb) [inline bin/download_proteins_entrez.py]:
```suggestion
        sys.exit("The format of the file provided with --taxid_input is invalid!")
```

File: bin/download_proteins_entrez.py [hunk 3]
```diff
@@ -119,37 +139,36 @@ def main(args=None):
 
     # 2) for each taxon -> select one assembly (largest for now)
     print("get assembly lengths and select largest assembly for each taxon ...")
-    dict_taxid_assemblyids = {}
+    dict_taxId_assemblyid = {} # TODO rename? only one assemlbyid per taxon
```
Comment (lkuchenb) [inline bin/download_proteins_entrez.py]: Remove TODO note?

File: bin/download_proteins_entrez.py [hunk 4]
```diff
@@ -169,16 +188,16 @@ def main(args=None):
     ### for each assembly get list of sequence ids
     # seq_ids = [ record["Id"] for assembly_record in nucleotide_results for record in assembly_record["LinkSetDb"][0]["Link"] ]
 
-    dict_seqid_assemblyids = {}
+    dict_seqId_assemblyIds = {}
```
Comment (lkuchenb) [inline bin/download_proteins_entrez.py]:
```suggestion
    from collections import defaultdict
    dict_seqId_assemblyIds = defaultdict(lambda : [])
```

File: bin/download_proteins_entrez.py [hunk 5]
```diff
@@ -169,16 +188,16 @@ def main(args=None):
...
     for assembly_record in nucleotide_results:
         assemblyid = assembly_record["IdList"][0]
         for record in assembly_record["LinkSetDb"][0]["Link"]:
-            if record["Id"] not in dict_seqid_assemblyids:
-                dict_seqid_assemblyids[record["Id"]] = [assemblyid]
+            if record["Id"] not in dict_seqId_assemblyIds:
+                dict_seqId_assemblyIds[record["Id"]] = [assemblyid]
             else:
```
Comment (lkuchenb) [inline bin/download_proteins_entrez.py]:
```suggestion

```

File: bin/download_proteins_entrez.py (hunk 4 above)
Comment (lkuchenb) [reply bin/download_proteins_entrez.py]: The two cases in the for loop can be avoided with a `defaultdict`

File: bin/download_proteins_entrez.py (hunk 1 above)
Comment (skrakau) [reply bin/download_proteins_entrez.py:85]: to go over all assemblies listed for the given protein ID, while in the next it goes over all tax IDs and assemblies and selects only the mapping ones...

File: bin/download_proteins_entrez.py (hunk 4 above)
Comment (skrakau) [reply bin/download_proteins_entrez.py]: thank you :)

**Pull request merged**